### PR TITLE
Issue #26: Add metadata wrapper objects with stub implementations

### PR DESCRIPTION
**Pull request by eemhu**
Contains changes:

group objects into more appropriate packages; 
remove parsedEvent() from Event interface; 
introduce wrapper objects for event metadata; 
change ParsedEvent interface to extend Event;
 add ParsedEventFactory decorator instead of each event decorating itself;
 add null checks to ParsedEventListFactory and use stub object if metadata null; 
add tests for null objects/ null values in arrays cases;
 add UnparsedEvent interface extending Event with parseJson() method;
 change EventImpl to UnparsedEventImpl

Fixes issue:
* #26

### CODE REVIEW

File: src/main/java/com/teragrep/akv_01/event/ParsedEventListFactory.java [hunk 1]
```diff
@@ -76,15 +98,43 @@ public List<ParsedEvent> asList() {
         final List<ParsedEvent> events = new ArrayList<>(payloads.length);
         for (int i = 0; i < payloads.length; i++) {
             if (payloads[i] != null) {
+                EventPartitionContext eventPartitionContext = eventPartitionContextStub;
+                EventProperties eventProperties = eventPropertiesStub;
+                EventSystemProperties eventSystemProperties = eventSystemPropertiesStub;
+                EnqueuedTime enqueuedTime = enqueuedTimeStub;
+                EventOffset eventOffset = eventOffsetStub;
+
+                if (partitionCtx != null) {
```
Comment (kortemik) [inline src/main/java/com/teragrep/akv_01/event/ParsedEventListFactory.java:107]: it would be better to use 
```
final EventPartitionContext partitionCtx;
if (partitionCtx != null) {
 partitionCtx = ...
}
else {
partitionCtx = stub
}

**Review by kortemik — APPROVED** [review]
